presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java

```diff
@@ -139,43 +145,22 @@ public synchronized void setResourceOvercommit()
         return future;
     }
 
-    public synchronized boolean tryReserveMemory(long bytes)
+    private synchronized boolean tryReserveMemory(long bytes)
     {
-        checkArgument(bytes >= 0, "bytes is negative");
-
-        if (reserved + bytes > maxMemory) {
+        if (queryMemoryContext.getUserMemory() + bytes > maxMemory) {
             return false;
         }
         if (memoryPool.tryReserve(queryId, bytes)) {
-            reserved += bytes;
             return true;
         }
         return false;
     }
 
-    public synchronized void freeMemory(long bytes)
+    public synchronized void freeUserMemory(long bytes)
```

Comment:
Remove this method, because the functionality is available in `updateUserMemory`